pkg/core/interfaces.go

```diff
@@ -128,3 +122,7 @@ type ExecutionManager interface {
 	// StoreCommandLogs stores the command logs in the azure.
 	StoreCommandLogs(ctx context.Context, blobPath string, reader io.Reader) <-chan error
 }
+
+type Requests interface {
```

Comment:
nit: add comment

Reply:
Addressed. However, this was supposed to be caught by automated `golangci-lint` checks. Its missing a configuration file. I have added that in this diff but eventually commented it out as all/any existing linting issues need to be fixed as a separate PR.